### PR TITLE
Remove deprecated numpy.float and numpy.bool

### LINKED ISSUE #255: Resolve deprecation warnings in tests
When running `make test` or `make test_ui`, pytest displays a bunch of deprecation warnings and some other warnings near the end of the test output. Nosetests probably wasn't catching these because it hasn't been updated in a long time. We should go through and resolve each issue causing a warning to prevent future problems. 

Most of them are pretty straightforward e.g. replacing `warn` with `warning`.

### PR DESCRIPTION
**Pull request by emlys**
# Description
Fixes #255 
> DeprecationWarning: `np.bool` is a deprecated alias for the builtin `bool`. To silence this warning, use `bool` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.bool_` here.

> DeprecationWarning: `np.float` is a deprecated alias for the builtin `float`. To silence this warning, use `float` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.float64` here.

Just replacing these.

### CODE REVIEW

File: src/natcap/invest/forest_carbon_edge_effect.py
Comment (dcdenu4) [inline src/natcap/invest/forest_carbon_edge_effect.py:845]: I think there are some comments above this call that reference `n_jobs`, could you update those?

Comment (dcdenu4) [reply src/natcap/invest/forest_carbon_edge_effect.py:845]: Also, does this mean we should set a lower bound on Scipy?

**Review by dcdenu4 — CHANGES_REQUESTED** [review]
Hey @emlys thanks for refactoring this. I noticed that a forest edge effect test was failing. It looks related to the `workers` keyword in the `query` op. Could this be a Scipy version issue?

**Comment by emlys** [comment]
Oh yep, it's `n_jobs` for older versions and `workers` for scipy>=1.6.0, the latest version. And it seems they're not phasing in/out the options like python tends to do. We would need to require scipy>=1.6.0 for this to work. Is that OK given that we've been allowing a wide range of scipy versions?
scipy>=1.6.0 doesn't break anything so maybe it's fine

**Comment by dcdenu4** [comment]
>  Is that OK given that we've been allowing a wide range of scipy versions?

Great question... I'd be curious what @phargogh thinks, but if the only reason to bump the lower bound for Scipy is to avoid this warning... I'd probably hold off on changing this for now. I wonder if there would be users who wouldn't necessarily want to be on the latest greatest Scipy version... But, like Rich said, who really are our python library users? Probably just us and our team.

**Comment by emlys** [comment]
It's unfortunate that the versions are mutually exclusive, but if we have to choose one, I'd lean towards the latest version. Have there been issues before with people unable to use the invest package due to requirements conflicts?

**Comment by dcdenu4** [comment]
Are the version mutually exclusive? If we left the `n_jobs` it just raises a deprecation warning no? But I could still use the latest version right?

I'm usually all about the latest and greatest and would agree with just bumping the version. But I feel like I normally miss something... and  @phargogh typically enlightens me haha

**Comment by emlys** [comment]
Ah, the warning says that `The old name "n_jobs" will stop working in SciPy 1.8.0.`. I guess I'd rather not wait until 1.8.0 comes out and the tests unexpectedly break, but we don't have to fix it right now either... I was trying to close #255 but it's really an eternal issue :/

**Comment by dcdenu4** [comment]
Yea, it looks like Scipy 1.6.0 just came out in late December. Hmmm. Well, the only reason I can think of to not go ahead and bump it is that users will not have upgraded to Scipy 1.6.0 because of it's recent release. There seems to be a lot of good changes in 1.6.0. Let's go for it!

**Comment by dcdenu4** [comment]
I think my only change then is the comments referring to `n_jobs` instead of `n_workers`

**Comment by emlys** [comment]
thanks @dcdenu4! Just updated those comments (removed one since it doesn't seem relevant any more)

**Comment by dcdenu4** [comment]
Great! I'll Approve once the tests go through.

**Review by dcdenu4 — APPROVED** [review]
Thanks!

**Comment by phargogh** [comment]
Yeah, I'm usually hesitant to upgrade the requirement for larger binary packages, but we shouldn't go further out of our way than really needed to maintain two parallel versions of a package without a clear use case benefiting from it.

Besides, it's gotten _so_ easy in the last few years to procure binary distributions of packages like `scipy` that requiring a later version shouldn't overall be an issue.

**Pull request merged**
